### ISSUE TITLE
Why ignore aes encrypt iv?

### ISSUE THREAD

**Issue by HeMinzhang** [dotnet/runtime#36730]
### Description

Why ignore aes encrypted iv?

such as [AesGcm](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.aesgcm?view=netcore-3.1)

### Configuration
![QQ图片20200520080812](https://user-images.githubusercontent.com/2683269/82390379-23279500-9a71-11ea-9c9d-9e83da2189a1.png)
![QQ图片20200520080753](https://user-images.githubusercontent.com/2683269/82390368-1dca4a80-9a71-11ea-8155-431100b26920.png)
![QQ图片20200520080807](https://user-images.githubusercontent.com/2683269/82390371-202ca480-9a71-11ea-8620-8421e3946467.png)

**Comment by vcsjones** [comment]
AES-GCM uses a nonce, which is similar but doesn't have the same properties as an IV which is used in AES-CBC. With `BCryptEncrypt`, the nonce gets passed in via the `BCRYPT_AUTHENTICATED_CIPHER_MODE_INFO`.

Your last screenshot doesn't seem to be for the right `BCryptEncrypt`, it's this one here;

https://github.com/dotnet/runtime/blob/07a5ee89b835a915ff3ac082ba3677fe39384a18/src/libraries/Common/src/Interop/Windows/BCrypt/Interop.BCryptEncryptDecrypt.cs#L72-L74.

In short:
1. AesGcm has a notion of a nonce, not an IV, though they serve a somewhat similar role, they have different properties.
2. CNG, when using AES-GCM, accepts the nonce, but not through the IV parameter, so it is null.

**Comment by HeMinzhang** [comment]
@vcsjones Thanks a million.
